### ISSUE TITLE
MA0011 should not be reported when all "template arguments" are case-insensitive

### ISSUE THREAD

**Issue by drieseng** [meziantou/Meziantou.Analyzer#1084]
Currently, **MA0011** (and probably other rules) is also reported when all of the "template arguments" are case-insensitive.

Example:
```c#
var sb = new StringBuilder();
_ = sb.AppendFormat("{0} / {1}", "X", "Y"); // MA0011 is reported
```
